goalie/go_mesh_seq.py

```diff
@@ -228,6 +232,17 @@ def indicate_errors(
                     indi.interpolate(abs(indi))
                     self.indicators[f][i][j].interpolate(ufl.max_value(indi, 1.0e-16))
 
+            # discard extra subinterval solution field
+            self.solutions[f][FWD_OLD].pop()
+            self.solutions[f][ADJ_NEXT].pop()
```

Comment:
Is there a situation in which we would not want to destroy these? If so, maybe add a kwarg to not destroy them?

And if we do destroy them, I think you should modify the `labels` in `self.solutions` at the end of `GoalOrientedMeshSeq.indicate_errors`, so that we don't run into errors trying to access data corresponding to those labels.

Reply:
@dundo - At the moment, after `indicate errors` runs through all subintervals, `self.solutions` still has a key for all default `labels` just with empty lists as associated values. Before destroying the label keys in `self.solutions`, can you think of a case where they might be reused after running `indicate errors`?

Reply:
I can't think of something like that, no - in the Hessian adaptation either. But I might be missing something, so I asked that question in the issue #222 :)

Reply:
@dundo - I've now modified `labels` and `solutions` dictionary to destroy the `forward_old` and `adjoint_next` categories in [4c0bc](https://github.com/mesh-adaptation/goalie/pull/224/commits/4c0bcb6e95660ff21df8acf953d61c82c557eeb2)